src/formatters.rs

```diff
@@ -0,0 +1,293 @@
+}
+
+/// Utility function to convert from polars `AnyValue` to `serde_json::Value`
+/// Doesn't cover all types but most of them.
+fn any_value_to_json(value: &AnyValue) -> Result<Value> {
+    match value {
+        AnyValue::Null => Ok(Value::Null),
+        AnyValue::Boolean(b) => Ok(Value::Bool(*b)),
+        AnyValue::String(s) => Ok(Value::String((*s).to_string())),
+        AnyValue::Int8(n) => Ok(json!(*n)),
+        AnyValue::Int16(n) => Ok(json!(*n)),
+        AnyValue::Int32(n) => Ok(json!(*n)),
+        AnyValue::Int64(n) => Ok(json!(*n)),
+        AnyValue::UInt8(n) => Ok(json!(*n)),
+        AnyValue::UInt16(n) => Ok(json!(*n)),
+        AnyValue::UInt32(n) => Ok(json!(*n)),
+        AnyValue::UInt64(n) => Ok(json!(*n)),
+        AnyValue::Float32(n) => Ok(json!(*n)),
+        AnyValue::Float64(n) => Ok(json!(*n)),
+        AnyValue::Date(d) => Ok(json!(d.to_string())), // You might want to format this
+        AnyValue::Datetime(dt, _, _) => Ok(json!(dt.to_string())), // You might want to format this
+        AnyValue::Time(t) => Ok(json!(t.to_string())), // You might want to format this
+        AnyValue::List(series) => {
+            let json_values: Result<Vec<Value>> =
+                series.iter().map(|val| any_value_to_json(&val)).collect();
+            Ok(Value::Array(json_values?))
+        }
+        _ => Err(anyhow!("Failed to convert type")),
+    }
+}
+
+/// Trait to define different output generators. Defines two
+/// functions, format which generates a serialized string of the
+/// `DataFrame` and save which generates a file with the generated
+/// file
+#[enum_dispatch]
+pub trait OutputGenerator {
+    fn save(&self, writer: &mut impl Write, df: &mut DataFrame) -> Result<()>;
+    fn format(&self, df: &mut DataFrame) -> Result<String> {
+        // Just creating an empty vec to store the buffered output
+        let mut data: Vec<u8> = vec![];
+        let mut buff = Cursor::new(&mut data);
+        self.save(&mut buff, df)?;
+
+        Ok(String::from_utf8(data)?)
+    }
+}
+
+/// Enum of OutputFormatters one for each potential
+/// output type
+#[enum_dispatch(OutputGenerator)]
+#[derive(Serialize, Deserialize, Debug)]
+pub enum OutputFormatter {
+    GeoJSON(GeoJSONFormatter),
+    GeoJSONSeq(GeoJSONSeqFormatter),
+    Csv(CSVFormatter),
+}
+
+/// Format the results as geojson sequence format
+/// This is one line per feature serialized as a
+/// geojson feature
+#[derive(Serialize, Deserialize, Debug)]
+pub struct GeoJSONSeqFormatter;
+
+impl OutputGenerator for GeoJSONSeqFormatter {
+    fn save(&self, writer: &mut impl Write, df: &mut DataFrame) -> Result<()> {
+        let geometry_col = df.column("geometry")?;
+        let other_cols = df.drop("geometry")?;
+        for (idx, geom) in geometry_col.str()?.into_iter().enumerate() {
+            if let Some(wkt_str) = geom {
+                let geom: Geometry<f64> = Geometry::try_from_wkt_str(wkt_str).unwrap();
```

Comment:
Could map the error with `anyhow!` (e.g. below). More generally wondering if it would be worth introducing a new `PopgetterError` enum type for writing conversions and fine-grained handling.
```suggestion
                let geom: Geometry<f64> = Geometry::try_from_wkt_str(wkt_str).map_err(|err| {
                    anyhow!("Invalid `Geometry<f64>` from well-known text string: {err}")
                })?;
```

Reply:
Sounds good. Like above we should never really have a condition where this fails but good to check.